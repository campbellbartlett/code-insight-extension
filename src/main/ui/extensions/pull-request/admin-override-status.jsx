import React from 'react';
import PropTypes from 'prop-types';

export const AdminOverrideStatus = ({ isOverride }) => (
    <div>
        {isOverride && (
            <h3>
                An administrator has provided authorisation to enable merging of this pull request
                before all code insight status reports have passed.
            </h3>
        )}
        {!isOverride && (
            <h3>
                This Pull Request cannot be merged until all code insight quality reports have
                passed.
            </h3>
        )}
    </div>
);

AdminOverrideStatus.propTypes = {
    isOverride: PropTypes.bool,
};

AdminOverrideStatus.defaultProps = {
    isOverride: false,
};
