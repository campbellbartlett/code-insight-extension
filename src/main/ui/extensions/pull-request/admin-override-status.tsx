import React from 'react';

type AdminOverrideStatusProps = {
    isOverride: boolean;
};

export const AdminOverrideStatus: React.FC<AdminOverrideStatusProps> = (
    props: AdminOverrideStatusProps
) => {
    const { isOverride } = props;
    return (
        <div id="admin-override-status">
            {isOverride ? (
                <h3 id="admin-message">
                    An administrator has provided authorisation to enable merging of this pull
                    request request before all code insight status reports have passed.
                </h3>
            ) : (
                <h3 id="standard-message">
                    This Pull Request cannot be merged until all code insight quality reports have
                    passed.
                </h3>
            )}
        </div>
    );
};
