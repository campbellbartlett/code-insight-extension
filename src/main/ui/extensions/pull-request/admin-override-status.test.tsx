import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';
import { AdminOverrideStatus } from './admin-override-status';

describe('admin override status renders correctly', () => {
    it('should render correctly', () => {
        const adminOverrideStatus = shallow(<AdminOverrideStatus isOverride />);

        expect(shallowToJson(adminOverrideStatus)).toMatchSnapshot();
    });

    it('should show correct message when admin', () => {
        const adminOverrideStatus = shallow(<AdminOverrideStatus isOverride />);

        expect(adminOverrideStatus.find('#admin-message').length).toEqual(1);
        expect(adminOverrideStatus.find('#standard-message').length).toEqual(0);
    });

    it('should show correct message when not an admin', () => {
        const adminOverrideStatus = shallow(<AdminOverrideStatus isOverride={false} />);

        expect(adminOverrideStatus.find('#admin-message').length).toEqual(0);
        expect(adminOverrideStatus.find('#standard-message').length).toEqual(1);
    });
});
