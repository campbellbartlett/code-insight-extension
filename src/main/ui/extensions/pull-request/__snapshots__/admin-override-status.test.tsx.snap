// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin override status renders correctly should render correctly 1`] = `
<div
  id="admin-override-status"
>
  <h3
    id="admin-message"
  >
    An administrator has provided authorisation to enable merging of this pull request request before all code insight status reports have passed.
  </h3>
</div>
`;
